### PR TITLE
Add TermFrequencyOffsetIterator and test

### PR DESCRIPTION
**Pull request by apmoriarty**
Adding alternate iterator for collecting term frequency offsets. Not integrated yet.

### CODE REVIEW

**Comment by apmoriarty** [comment]
This is just adding code, not integrating it.

File: warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+ */
+public class TermFrequencyOffsetIterator implements SortedKeyValueIterator<Key,Value> {
+    
+    private static final Logger log = Logger.getLogger(TermFrequencyOffsetIterator.class);
+    
+    private Key tk = null;
+    private Value tv = null;
+    
+    // A set of full column qualifiers that describe the offset's exact location (datatype\0uid\0value\0FIELD)
+    private final SortedSet<Text> searchSpace;
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java]: I am wondering why we need this iterator.  This is essentially equivalent to supplying a set of ranges with the underlying source that include only the searchSpace values.  If we truly need this as as separate iterator, then I would point you to consider extending accumulo's SeekingFilter instead.

**Review by ivakegg — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java]: I added a TermFrequencyFilter offered up as an alternative implementation

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java]: Great work -- I removed the TermFrequencyOffsetIter in the latest commit in favor of the TermOffsetFilter

**Review by ivakegg — APPROVED** [review]

**Review by avgAGB — APPROVED** [review]

**Pull request merged**